### PR TITLE
Run the update command to update each and every single extension.

### PR DESCRIPTION
**Pull request by alexventuraio**
Following this comment https://github.com/open-vsx/publish-extensions/pull/155#pullrequestreview-485350703 and running the suggested command:

```sh
node upgrade-extensions
```

I got this result, so @jankeromnes can you take a look and comment which of the updates truly apply for the update, please?

### CODE REVIEW

**Comment by jankeromnes** [comment]
Awesome, many thanks @alexventuraio!

I'll take a look at the result. While this command automates most of the work, it still has a few corner cases that need manual adjustments. Will comment in-line in the PR where relevant.

File: extensions.json [hunk 1]
```diff
@@ -455,8 +455,8 @@
     {
       "id": "hediet.vscode-drawio",
       "repository": "https://github.com/hediet/vscode-drawio",
-      "version": "0.7.2",
-      "checkout": "v0.7.2",
+      "version": "999.0.0-alpha",
+      "checkout": "master",
```
Comment (jankeromnes) [inline extensions.json:459]: The latest release tag from that repo is still `v0.7.2`.

However, even under that tag, the `package.json` `version` is indeed `999.0.0-alpha`, which caused the OpenVSX upload to have that version as well: https://open-vsx.org/extension/hediet/vscode-drawio

It seems like we're missing some kind of `"prepublish"` build or packaging or versioning command.

Looking at [the `package.json` "scripts"](https://github.com/ayu-theme/vscode-ayu/blob/48eab0dc3d9ea4bcf209cd355337977ddbd7a5d1/package.json#L70-L75), I think we should run `npm run build` and/or `npm run package` in our `"prepublish"` step.

@spoenemann Do you agree?

File: extensions.json [hunk 2]
```diff
@@ -794,14 +794,14 @@
...
+      "download": "https://github.com/Microsoft/vscode-python/releases/download/2020.9.112786/ms-python-release.vsix",
+      "version": "2020.9.112786"
     },
     {
       "id": "ms-vscode.atom-keybindings",
       "repository": "https://github.com/Microsoft/vscode-atom-keybindings",
-      "version": "3.0.6",
-      "checkout": "v3.0.6"
+      "version": "3.0.9",
+      "checkout": "master"
```
Comment (jankeromnes) [inline extensions.json:804]: The latest release tag from that repo is still `v3.0.6`. Please undo this change.

File: extensions.json [hunk 3]
```diff
@@ -794,14 +794,14 @@
     },
     {
       "id": "ms-python.python",
-      "download": "https://github.com/Microsoft/vscode-python/releases/download/2020.9.111407/ms-python-release.vsix",
-      "version": "2020.9.111407"
+      "download": "https://github.com/Microsoft/vscode-python/releases/download/2020.9.112786/ms-python-release.vsix",
+      "version": "2020.9.112786"
```
Comment (jankeromnes) [inline extensions.json:798]: 👍

File: extensions.json [hunk 4]
```diff
@@ -818,23 +818,22 @@
     {
       "id": "ms-vscode.js-debug",
       "repository": "https://github.com/microsoft/vscode-js-debug",
-      "version": "1.50.0",
-      "checkout": "v1.50.0",
-      "prepublish": "npm run compile && npm run package",
-      "extensionFile": "dist/js-debug.vsix"
+      "version": "1.50.1",
+      "checkout": "master",
```
Comment (jankeromnes) [inline extensions.json:822]: The latest release tag seems to be `v1.50.2`:

```suggestion
      "checkout": "v1.50.2",
```

File: extensions.json [hunk 5]
```diff
@@ -818,23 +818,22 @@
...
+      "prepublish": "npm run compile && npm run package"
```
Comment (jankeromnes) [inline extensions.json:823]: Oops, the upgrade script stripped `"extensionFile": "dist/js-debug.vsix"` from the extension parameters.

That's because this parameter is quite new, and we forgot to make `upgrade-extensions.js` explicitly keep it.

Could you please make `upgrade-extensions.js` keep the `extensionFile` parameter by:
1. Adding a `--extensionFile` CLI flag in [add-extension.js](https://github.com/open-vsx/publish-extensions/blob/master/add-extension.js) (e.g. everywhere you see the `prepublish` CLI option being used)
2. Adding the `--extensionFile` CLI flag in `upgrade-extensions.js` when the parameter exists:

https://github.com/open-vsx/publish-extensions/blob/6dcbd4864d9b8bf9b0c6384be0f79e1f77cf976d/upgrade-extensions.js#L50-L59

Also, please manually add `extensionFile` back here before we forget about it 😅:

```suggestion
      "prepublish": "npm run compile && npm run package",
      "extensionFile": "dist/js-debug.vsix"
```

File: extensions.json [hunk 6]
```diff
@@ -818,23 +818,22 @@
...
     },
     {
       "id": "ms-vscode.node-debug",
       "repository": "https://github.com/microsoft/vscode-node-debug",
-      "version": "1.44.8",
-      "checkout": "v1.44.8",
+      "version": "1.44.14",
+      "checkout": "master",
```
Comment (jankeromnes) [inline extensions.json:829]: The latest release tag still seems to be `v1.44.8`. Please undo this change.

File: extensions.json [hunk 7]
```diff
@@ -818,23 +818,22 @@
...
       "prepublish": "yarn prepublish && yarn package"
     },
     {
       "id": "mtxr.sqltools",
       "repository": "https://github.com/mtxr/vscode-sqltools",
-      "version": "0.23.0",
-      "checkout": "v0.23.0"
+      "version": "1.0.0",
+      "checkout": "master"
```
Comment (jankeromnes) [inline extensions.json:836]: This extension has very confusing release tags.

The latest version published to the Microsoft marketplace is `0.23.0`: https://marketplace.visualstudio.com/items?itemName=mtxr.sqltools

However, there is no `0.23.0` or `v0.23.0` tags in the repo. There is `v0.22.11`, which was last modified in July, and also `v1.2.0` which was last modified in May?

I have no idea what's going on here. Probably safer to just undo this upgrade.

File: extensions.json [hunk 8]
```diff
@@ -863,8 +862,8 @@
     {
       "id": "nwolverson.ide-purescript",
       "repository": "https://github.com/nwolverson/vscode-ide-purescript",
-      "version": "0.23.0",
-      "checkout": "v0.23.0",
+      "version": "0.23.1",
+      "checkout": "master",
```
Comment (jankeromnes) [inline extensions.json:866]: Latest release tag is still `0.23.0`. Please undo this change.

File: extensions.json [hunk 9]
```diff
@@ -1027,8 +1026,8 @@
     {
       "id": "streetsidesoftware.code-spell-checker",
       "repository": "https://github.com/streetsidesoftware/vscode-spell-checker",
-      "version": "1.9.2",
-      "checkout": "v1.9.2",
+      "version": "1.9.3-alpha.0",
+      "checkout": "master",
```
Comment (jankeromnes) [inline extensions.json:1030]: The latest release tag still seems to be `v1.9.2`. Please undo this change.

File: extensions.json [hunk 10]
```diff
@@ -1049,8 +1048,8 @@
     {
       "id": "timonwong.shellcheck",
       "repository": "https://github.com/timonwong/vscode-shellcheck",
-      "version": "0.10.1",
-      "checkout": "v0.10.1"
+      "version": "0.11.0",
+      "checkout": "master"
```
Comment (jankeromnes) [inline extensions.json:1052]: The latest release tag still seems to be `v0.10.1`. Please undo this change.

**Review by jankeromnes — COMMENTED** [review]
Many thanks for running a batch-upgrade!

Unfortunately, as I mentioned, the script isn't fully-automated yet, and still requires a bit of manual cleanup.

I've replied line-by-line to point out what we can keep and what should be adjusted. We've also identified a few minor improvements we could bring to `upgrade-extensions.js`. Would you have time to implement them? That would be super helpful. 💯

File: extensions.json [hunk 11]
```diff
@@ -214,14 +214,14 @@
...
+      "version": "3.16.0-dev",
+      "checkout": "master"
     },
     {
       "id": "Dart-Code.flutter",
       "repository": "https://github.com/Dart-Code/Flutter",
-      "version": "3.14.1",
-      "checkout": "v3.14.1"
+      "version": "3.15.0",
+      "checkout": "master"
```
[the remark this answers is not in the document]
Comment (alexventuraio) [reply extensions.json]: Done!

File: extensions.json [hunk 12]
```diff
@@ -252,8 +252,8 @@
     {
       "id": "DigitalBrainstem.javascript-ejs-support",
       "repository": "https://github.com/Digitalbrainstem/ejs-grammar",
-      "version": "1.2.1",
-      "checkout": "v1.2.1"
+      "version": "1.3.1",
+      "checkout": "master"
```
[the remark this answers is not in the document]
Comment (alexventuraio) [reply extensions.json]: Done!

File: extensions.json (hunk 2 above)
Comment (alexventuraio) [reply extensions.json]: Done!

File: extensions.json (hunk 4 above)
Comment (alexventuraio) [reply extensions.json]: Done!

File: extensions.json (hunk 6 above)
Comment (alexventuraio) [reply extensions.json]: Done!

File: extensions.json (hunk 7 above)
Comment (alexventuraio) [reply extensions.json]: Done!

File: extensions.json (hunk 8 above)
Comment (alexventuraio) [reply extensions.json]: Done!

File: extensions.json (hunk 10 above)
Comment (alexventuraio) [reply extensions.json]: Done!

File: extensions.json (hunk 9 above)
Comment (alexventuraio) [reply extensions.json]: done!

File: extensions.json [hunk 13]
```diff
@@ -214,14 +214,14 @@
     {
       "id": "Dart-Code.dart-code",
       "repository": "https://github.com/Dart-Code/Dart-Code",
-      "version": "3.14.1",
-      "checkout": "v3.14.1"
+      "version": "3.16.0-dev",
+      "checkout": "master"
```
[the remark this answers is not in the document]
Comment (alexventuraio) [reply extensions.json]: Done!

File: extensions.json (hunk 5 above)
Comment (alexventuraio) [reply extensions.json]: @jankeromnes Please take a look at the changes I did in order to make sure I introduced the correct CLI tag in an expected way!

File: extensions.json (hunk 1 above)
Comment (spoenemann) [reply extensions.json:459]: The package script just runs `vsce package`, which is already done by `ovsx`.

Comment (jankeromnes) [reply extensions.json:459]: Wait sorry, that was the wrong `package.json` file. [Here is the correct one](https://github.com/hediet/vscode-drawio/blob/14209ad462028cbf7013bc8e094bf0fc6e811d30/package.json#L272-L280).

@spoenemann so, do you think we should add anything else to our `"prepublish"` step (which already has `yarn pre-build`)?

Meanwhile, maybe let's undo these changes:

```suggestion
      "version": "0.7.2",
      "checkout": "v0.7.2",
```

File: extensions.json (hunk 5 above)
Comment (jankeromnes) [reply extensions.json]: @alexventuraio Awesome! Your changes look good to me. 👍

Many thanks for taking the time to contribute this. 💯

**Review by jankeromnes — APPROVED** [review]
Apart from the changes to `hediet.vscode-drawio`, this Pull Request looks good to me!

Many thanks again @alexventuraio for working on this. 🙏 

Let's resolve the DrawIO version/checkout/prepublish item, and merge this PR. 🚀

File: extensions.json (hunk 5 above)
Comment (alexventuraio) [reply extensions.json]: All a pleasure to help something I like and find it useful for me! 😃

File: extensions.json (hunk 1 above)
Comment (alexventuraio) [reply extensions.json]: @jankeromnes Done!

**Comment by alexventuraio** [comment]
I completed the changes you mentioned, I think everything is done!
Waiting for this PR to be merged!

And maybe one more thing, do you guys think it could be possible to label the **accepted and merged** PRs with the label `hacktoberfest-accepted` as [mentioned here](https://hacktoberfest.digitalocean.com/hacktoberfest-update?utm_medium=email&utm_source=hacktoberfest&utm_campaign=main_&utm_content=response) in order for them to count as a valid PR for the contributors so we can get the swag from the HacktoberFest contest?

Thanks in advance 🙏

**Comment by alexventuraio** [comment]
> Many thanks for running a batch-upgrade!
> 
> Unfortunately, as I mentioned, the script isn't fully-automated yet, and still requires a bit of manual cleanup.
> 
> I've replied line-by-line to point out what we can keep and what should be adjusted. We've also identified a few minor improvements we could bring to `upgrade-extensions.js`. Would you have time to implement them? That would be super helpful. 💯

Algo @jankeromnes we can talk about the updates/improvements you want to implement in the `upgrade-extensions.js`. I would be pleased to work on it.

**Comment by jankeromnes** [comment]
> I completed the changes you mentioned, I think everything is done!
> Waiting for this PR to be merged!

Awesome, many thanks! I confirm that everything looks resolved and will thus merge this PR.

> And maybe one more thing, do you guys think it could be possible to label the accepted and merged PRs with the label hacktoberfest-accepted as mentioned here in order for them to count as a valid PR for the contributors so we can get the swag from the HacktoberFest contest?

For sure! Thanks for pointing this out. I see that @spoenemann has already created the label, and I'll add it to this PR.

> Also, @jankeromnes we can talk about the updates/improvements you want to implement in the upgrade-extensions.js. I would be pleased to work on it.

Actually, I believe that you've already implemented the biggest items! (I.e. supporting `extensionFile` and ignoring `-insider` releases.) Thanks again for that.

There are probably further improvements we could make, e.g. to reduce the number of false positives & manual work required after running `node upgrade-extensions`.

One idea would be for `"checkout"` upgrades to try and pick the latest available release tag in the repo, as opposed to just picking the default branch, and @niklaskeerl actually already wrote a script that does that, which we could potentially integrate, see https://github.com/open-vsx/publish-extensions/pull/168#issuecomment-700727265. Another idea would be to check for `.vsix` releases for all extensions (i.e. with `"download"` URL instead of `"repository"`), and upgrade to that if it's available, because that's always safer / more robust than trying to re-package extensions ourselves from repositories / tags.
